### PR TITLE
Support registration of 6x RGB.

### PR DESCRIPTION
**Pull request by SamCWill**
Allow the user to register RGB images and to shave off edges of registered images such that only the pixels shared by all registered channels remain.

### CODE REVIEW

File: imgreg/multi_spect_image_io.py [hunk 1]
```diff
@@ -1,53 +1,132 @@
-# multi_spect_image_io.py
-# Copyright (c) 2020, Kostas Alexis, Frank Mascarich, University of Nevada, Reno.
-# All rights reserved.
+"""
+Image input/output handling.
+
+multi_spect_image_io.py
+Copyright (c) 2020, Kostas Alexis, Frank Mascarich, University of Nevada, Reno.
+All rights reserved.
+"""
```
Comment (dzermas) [inline imgreg/multi_spect_image_io.py:7]: @SamCWill Add your name and Sentera as an organization. Something like 
```Code revision 2021, Sam Williams, Sentera Inc.```

File: imgreg/multi_spect_plotting_utils.py [hunk 2]
```diff
@@ -1,95 +1,137 @@
-# multi_spect_plotting_utils.py
-# Copyright (c) 2020, Kostas Alexis, Frank Mascarich, University of Nevada, Reno.
-# All rights reserved.
+"""
+Display images and metrics.
+
+multi_spect_plotting_utils.py
+Copyright (c) 2020, Kostas Alexis, Frank Mascarich, University of Nevada, Reno.
+All rights reserved.
```
Comment (dzermas) [inline imgreg/multi_spect_plotting_utils.py:6]: @SamCWill Same here

File: multi_spect_reg.py [hunk 3]
```diff
@@ -1,7 +1,12 @@
-# multi_spect_reg.py
-# Copyright (c) 2020, Kostas Alexis, Frank Mascarich, University of Nevada, Reno.
-# All rights reserved.
+"""
+Perform multi-spectral image registration.
+
+multi_spect_reg.py
+Copyright (c) 2020, Kostas Alexis, Frank Mascarich, University of Nevada, Reno.
+All rights reserved.
```
Comment (dzermas) [inline multi_spect_reg.py:6]: @SamCWill and here...

File: imgreg/multi_spect_reg_config.py [hunk 4]
```diff
@@ -1,167 +1,260 @@
...
+                    self.image_extension = ".tif"
+            else:
+                print(
+                    f"Image file {file_list[0]} is not of a supported type. (.jpg, ,tif, .tiff)"
+                )
+                raise TypeError
+            for file_name in file_list:
+                img_str_list = list(file_name.replace(".", "_").split("_"))
+                for img_str in img_str_list:
+                    if img_str.isdigit:
```
Comment (dzermas) [inline imgreg/multi_spect_reg_config.py]: This should be a method, right? `img_str.isdigit()`

**Comment by dzermas** [comment]
This is some dope utility! The multispectral bands are aligned nicely. The RGB's scale is a tiny bit larger than the single bands though and it makes it look out of alignment. What assumptions are you making for the RGB when resizing?

**Review by dzermas — CHANGES_REQUESTED** [review]
We need to find why the output RGB ends up larger than the multispectral ones.

File: imgreg/multi_spect_reg_config.py (hunk 4 above)
Comment (SamCWill) [reply imgreg/multi_spect_reg_config.py]: I forgot to push this! Good catch.

**Comment by SamCWill** [comment]
> We need to find why the output RGB ends up larger than the multispectral ones.

I think this is because the RGB camera has a wider field of view than the single-band sensors. I think this could probably be fixed by just shaving some fixed number of pixels from the edges of the RGB image when it's read in. I'll figure this out.

**Comment by dzermas** [comment]
> > We need to find why the output RGB ends up larger than the multispectral ones.
> 
> I think this is because the RGB camera has a wider field of view than the single-band sensors. I think this could probably be fixed by just shaving some fixed number of pixels from the edges of the RGB image when it's read in. I'll figure this out.

I don't think shaving pixels off will solve this. It is a scaling issue. The RGB's scale is a tiny bit larger than the multispectral's.

**Review by dzermas — APPROVED** [review]
No major issues.

**Pull request merged**